qa/rpc-tests/feature_issuance.py

```diff
@@ -160,5 +236,88 @@ def run_test(self):
+        process_raw_issuance(self.nodes[0], [{"asset_amount":7, "asset_address":blind_addr, "token_amount":2, "token_address":nonblind_addr, "blind":False}, {"asset_amount":2, "asset_address":nonblind_addr, "blind":False}])
+        # Up to 5 issuances since we're making 5 outputs each time
+        process_raw_issuance(self.nodes[0], [{"asset_amount":7, "asset_address":nonblind_addr, "token_amount":2, "token_address":blind_addr, "blind":False}, {"asset_amount":2, "asset_address":nonblind_addr, "blind":False}])
+        process_raw_issuance(self.nodes[0], [{"asset_amount":1, "asset_address":nonblind_addr, "token_amount":2, "token_address":blind_addr, "blind":False}, {"asset_amount":3, "asset_address":nonblind_addr, "blind":False}, {"asset_amount":4, "asset_address":nonblind_addr, "token_amount":5, "token_address":blind_addr, "blind":False}, {"asset_amount":6, "asset_address":nonblind_addr, "token_amount":7, "token_address":blind_addr, "blind":False}, {"asset_amount":8, "asset_address":nonblind_addr, "token_amount":9, "token_address":blind_addr, "blind":False}])
+        # Default "blind" value is true, ommitting explicit argument for last
+        process_raw_issuance(self.nodes[0], [{"asset_amount":1, "asset_address":nonblind_addr, "token_amount":2, "token_address":blind_addr, "blind":True}, {"asset_amount":3, "asset_address":nonblind_addr, "blind":True}, {"asset_amount":4, "asset_address":nonblind_addr, "token_amount":5, "token_address":blind_addr, "blind":True}, {"asset_amount":6, "asset_address":nonblind_addr, "token_amount":7, "token_address":blind_addr, "blind":True}, {"asset_amount":8, "asset_address":nonblind_addr, "token_amount":9, "token_address":blind_addr}])
+
+        # Make sure contract hash is being interpreted as expected, resulting in different asset ids
+        raw_tx = self.nodes[0].createrawtransaction([], {nonblind_addr:self.nodes[0].getbalance()['bitcoin']-1})
+        funded_tx = self.nodes[0].fundrawtransaction(raw_tx)['hex']
+        id_set = set()
+
+        # First issue an asset with no argument
+        issued_tx = self.nodes[2].rawissueasset(funded_tx, [{"asset_amount":1, "asset_address":nonblind_addr}])[0]["hex"]
+        decode_tx = self.nodes[0].decoderawtransaction(issued_tx)
+        id_set.add(decode_tx["vin"][0]["issuance"]["asset"])
+
+        # Again with "" argument, which shouldn't be hashed but parsed as no argument at all
+        issued_tx = self.nodes[2].rawissueasset(funded_tx, [{"asset_amount":1, "asset_address":nonblind_addr, "contract":""}])[0]["hex"]
+        decode_tx = self.nodes[0].decoderawtransaction(issued_tx)
+        id_set.add(decode_tx["vin"][0]["issuance"]["asset"])
+        assert_equal(len(id_set), 1)
+
+        # Random contract string should again differ
+        issued_tx = self.nodes[2].rawissueasset(funded_tx, [{"asset_amount":1, "asset_address":nonblind_addr, "contract":"deadbeef"}])[0]["hex"]
```

Comment:
Reminder: contract value 32-bytes.

Reply:
fixed in other PR, will rebased post-merge